src/recaplog

```diff
@@ -20,145 +20,252 @@
+    matched_files=$(
+      find ${BASEDIR} -maxdepth 2 -regextype posix-extended \
+        -type f -mtime +${LOG_EXPIRY} -regex \
+        "^${BASEDIR}/(.*/)?[a-z]+_([a-z]+_)?[0-9]{8}(-[0-9]{6})?\.log(\.tar)?(\.gz)?$"
+    )
+    if [[ -n ${matched_files} ]]; then
+      count_files=$( wc -l <<<"${matched_files}" )
+    fi
+    log INFO "${prefix_msg} ${count_files} log files."
+    if [[ ! ${DRYRUN} ]]; then
+      # Deleting old logs, compressed/tarballs and empty dirs.
+      for matched_file in ${matched_files}; do
+        rm -f ${matched_file}
+      done
+      # Deleting empty dirs generated by recaplog when no compressing logs.
+      empty_dirs=$( find ${BASEDIR} -maxdepth 1 -regextype posix-extended \
+                      -regex "^${BASEDIR}/[a-z]+_[a-z]+_[0-9]{8}$" -empty \
+                      -type d )
+      if [[ -n ${empty_dirs} ]]; then
+        count_dirs=$( wc -l <<<"${empty_dirs}" )
+      fi
+      log INFO "${prefix_msg} ${count_dirs} empty directories."
+      for empty_dir in ${empty_dirs}; do
+        rmdir "${empty_dir}"
+      done
+    fi
+  fi
 }
 
-# cat tiny log files into daily log files
+# Function to compress/move daily log files
+# Previously the logs were concatenated and could be compressed, that is a
+# problem for parsing them at a later time as reported in:
+# https://github.com/rackerlabs/recap/pull/28
+# The new behaviour is to move the log files to a directory then opt for
+# compression, if compressed then it will delete the original logs.
 pack_old_logs() {
-	YESTERDAYDATE=$(date -d "now -1 day" "+%Y%m%d")
-	for item in $COMPRESS_ITEMS; do
-		print_msg DEBUG "packing ${item}..."
-		OUTPUTFILE=${BASEDIR}/${item}_daily_${YESTERDAYDATE}.log
-		#
-		if [ $LOG_COMPRESS -eq 1 ]; then
-			print_msg DEBUG "do compress old log files"
-			cat ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log | gzip > ${OUTPUTFILE}.gz
-			ERRCODE=$?
-			touch -t ${YESTERDAYDATE}0000 $OUTPUTFILE.gz
-		else
-			print_msg DEBUG "do not compress old log files"
-			cat ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log > $OUTPUTFILE
-			ERRCODE=$?
-			touch -t ${YESTERDAYDATE}0000 $OUTPUTFILE
-		fi
-		if [ $ERRCODE -eq 0 ]; then
-			for f in $(ls ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log | grep -v daily); do
-				print_msg DEBUG "deleting: $f"
-				rm -f $f
-			done 
-		else
-			print_msg ERROR "cat error, cannot delete old log files '"${BASEDIR}/${item}_${YESTERDAYDATE}-*.log"'"
-		fi
-	done
+  local -a errors
+  local YESTERDAYDATE=$( date -d "now -1 day" "+%Y%m%d" )
+  if [[ ${LOG_COMPRESS} -eq 1 ]]; then
+    log INFO "Compressing old log files"
+  else
+    log INFO "Not compressing old log files"
+  fi
+  # Backing up only items defined in the config
+  for item in ${BACKUP_ITEMS}; do
+    YESTERDAY_ITEM_DIR="${BASEDIR}/${item}_daily_${YESTERDAYDATE}"
+    err=''
+    log INFO "Packing ${item}..."
+    log_count=$(
+      {
+        ls -1 ${BASEDIR}/${item}_${YESTERDAYDATE}-*.log | wc -l
+      } 2>&1; exit ${PIPESTATUS[0]}
```

Comment:
I'm not a fan of this structure - particularly the use of 'exit'.  At a glance, it seems as if execution halts there, but we're really just capturing a variable.  Also, I've had issues in the past trying to parse the output of 'ls'.
What about this instead?
```
find "${BASEDIR}" -type f -name "${item_${YESTERDAYDATE}-\*.log" 2>&1 | wc -l
```
Then, instead of the next 'if' testing $?, just specifically test ${PIPESTATUS[0]}.